File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool check(string s, string t){
        unordered_map<char,char> mp;
        int n = s.size();
        for(int i = 0; i<n; i++){
            if(mp.find(s[i]) == mp.end()){
                mp.insert({s[i],t[i]});
            }else{
                if(mp[s[i]] != t[i]) return false;
            }
        }
        return true;
    }
    bool isIsomorphic(string s, string t) {
        if(s.size() != t.size()) return false;
        if(check(s,t) && check(t,s)) return true;
        return false;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    const check = function(s,t){
        const mp = new Map();
        for(let i = 0; i<s.length; i++){
            if(mp.has(s[i])){
                if(mp.get(s[i]) != t[i]) return false
            }else{
                mp.set(s[i],t[i]);
            }
        }
        return true;
    }
    if(s.length != t.length) return false
    if(check(s,t) && check(t,s)) return true;
    return false;
};